### PR TITLE
[FIX #7777, #7776]: Generate valid code if comment exists before closing brace

### LINKED ISSUE #7777: Rubocop places commas into comments, generating invalid ruby code
When formatting the following code, rubocop will generate invalid ruby:

```ruby
class Bar
  def initialize(*args)
  end
end

class Foo < Bar
  def initialize
    super(
      'firstKey' => [ 'value a', #description1
                      'value b' #description2
      ],
      'secondKey' => 'Bar'
    )
  end
end
```

## Expected behavior

Not to generate invalid ruby code

## Actual behavior

Invalid ruby code is created due to incorrectly handling the comment and placing a comma on the comment line:

```ruby
class Bar
  def initialize(*args)
  end
end

class Foo < Bar
  def initialize
    super(
      'firstKey' => ['value a', # description1
                      'value b'] # description2,
      'secondKey' => 'Bar'
    )
  end
end
```

## Steps to reproduce the problem

1. Create new ruby file
2. add original code snippet
3. bundle exec rubocop  -x --safe-auto-correct new_file.rb
4. Run again, verify invalid ruby code has been generated

## RuboCop version

```
$ [bundle exec] rubocop -V
0.80.0 (using Parser 2.7.0.2, running on ruby 2.6.5 x86_64-darwin18)
```

### PR DESCRIPTION
**Pull request by shekhar-patil**
closes #7777 and #7776 

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cli/cli_autocorrect_spec.rb [hunk 1]
```diff
@@ -945,6 +945,44 @@ class Bar
       RUBY
   end
 
+  it 'correct multiline array brace without crashing' do
```
Comment (mwolman) [inline spec/rubocop/cli/cli_autocorrect_spec.rb:948]:
```suggestion
  it 'corrects multiline array brace without crashing' do
```

File: spec/rubocop/cli/cli_autocorrect_spec.rb [hunk 2]
```diff
@@ -945,6 +945,44 @@ class Bar
...
+    create_file('example.rb', <<~RUBY)
+      hash = {
+        first_key: [ 'value a', # comment 1
+                  'value b'  # comment 2
+        ],
+        second_key: 'value c'
+      }
+    RUBY
+    expect(cli.run(['--auto-correct'])).to eq(1)
```
Comment (mwolman) [inline spec/rubocop/cli/cli_autocorrect_spec.rb:957]:
```suggestion
    
    expect(cli.run(['--auto-correct'])).to eq(1)
```

File: spec/rubocop/cli/cli_autocorrect_spec.rb [hunk 3]
```diff
@@ -945,6 +945,44 @@ class Bar
...
+
+        hash = {
+          first_key: ['value a', # comment 1
+                      'value b'], # comment 2
+          second_key: 'value c'
+        }
+      RUBY
+  end
+
+  it 'correct multiline literal brace without crashing' do
```
Comment (mwolman) [inline spec/rubocop/cli/cli_autocorrect_spec.rb:970]:
```suggestion
  it 'corrects multiline literal brace without crashing' do
```

File: spec/rubocop/cli/cli_autocorrect_spec.rb [hunk 4]
```diff
@@ -945,6 +945,44 @@ class Bar
...
+    create_file('example.rb', <<~RUBY)
+      super(foo(bar,
+        'hash' => {'key' => 'value'} # comment
+      ))
+    RUBY
+    expect(cli.run(['--auto-correct'])).to eq(0)
```
Comment (mwolman) [inline spec/rubocop/cli/cli_autocorrect_spec.rb:976]:
```suggestion
    
    expect(cli.run(['--auto-correct'])).to eq(0)
```

File: spec/rubocop/cli/cli_autocorrect_spec.rb (hunk 1 above)
Comment (shekhar-amura) [reply spec/rubocop/cli/cli_autocorrect_spec.rb]: Done.

File: spec/rubocop/cli/cli_autocorrect_spec.rb (hunk 2 above)
Comment (shekhar-amura) [reply spec/rubocop/cli/cli_autocorrect_spec.rb:958]: Done.

File: spec/rubocop/cli/cli_autocorrect_spec.rb (hunk 3 above)
Comment (shekhar-amura) [reply spec/rubocop/cli/cli_autocorrect_spec.rb]: Done.

File: spec/rubocop/cli/cli_autocorrect_spec.rb (hunk 4 above)
Comment (shekhar-amura) [reply spec/rubocop/cli/cli_autocorrect_spec.rb:978]: Done.

Comment (shekhar-patil) [reply spec/rubocop/cli/cli_autocorrect_spec.rb:978]: Done.

File: spec/rubocop/cli/cli_autocorrect_spec.rb (hunk 3 above)
Comment (shekhar-patil) [reply spec/rubocop/cli/cli_autocorrect_spec.rb]: Done.

File: spec/rubocop/cli/cli_autocorrect_spec.rb (hunk 2 above)
Comment (shekhar-patil) [reply spec/rubocop/cli/cli_autocorrect_spec.rb:958]: Done.

File: spec/rubocop/cli/cli_autocorrect_spec.rb (hunk 1 above)
Comment (shekhar-patil) [reply spec/rubocop/cli/cli_autocorrect_spec.rb]: Done.

File: spec/rubocop/cli/cli_autocorrect_spec.rb [hunk 5]
```diff
@@ -945,6 +945,46 @@ class Bar
       RUBY
   end
 
+  it 'corrects multiline array brace without crashing' do
```
Comment (bbatsov) [inline spec/rubocop/cli/cli_autocorrect_spec.rb:948]: Why didn't you add those tests to the spec of the cop instead? It seems to me this is the natural place for them.

Comment (shekhar-patil) [reply spec/rubocop/cli/cli_autocorrect_spec.rb]: Done.

File: lib/rubocop/cop/correctors/multiline_literal_brace_corrector.rb [hunk 6]
```diff
@@ -42,8 +42,30 @@ def correct_next_line_brace(corrector)
 
         corrector.insert_before(
           last_element_range_with_trailing_comma(node).end,
+          closing_brace_content(corrector, node)
+        )
+      end
+
+      def closing_brace_content(corrector, node)
```
Comment (bbatsov) [inline lib/rubocop/cop/correctors/multiline_literal_brace_corrector.rb:49]: I think the names `closing_brace_content` and `trailing_content_of_closing_brace` are a bit confusing, as braces can't really have content. The naming should be improved.

Comment (shekhar-amura) [reply lib/rubocop/cop/correctors/multiline_literal_brace_corrector.rb]: @bbatsov changes done. Please review now.

**Comment by shekhar-patil** [comment]
@bbatsov this is kind reminder to you. Please review and let me know if any changes required.

**Pull request closed without merging**